crates/compiler/mono/src/code_gen_help/equality.rs

```diff
@@ -340,33 +346,46 @@ fn eq_tag_union_help<'a>(
     let mut tag_branches = Vec::with_capacity_in(tag_layouts.len(), root.arena);
 
     // If there's a null tag, check it first. We might not need to load any data from memory.
-    if let Some(id) = nullable_id {
+    if let NullableId::Wrapped(id) = nullable_id {
         tag_branches.push((id as u64, BranchInfo::None, Stmt::Ret(Symbol::BOOL_TRUE)))
+    } else if let NullableId::Unwrapped(id) = nullable_id {
```

Comment:
Oh I didn't know you could do `else if let`.
It does feel like a `match` would be nicer than destructuring the same thing twice though.

Reply:
@JTeeuwissen can you fix that on your branch. you'll have to merge this in anyway